### DISCUSSION TITLE
How to export the indexQuery to csv

### DISCUSSION THREAD

**Discussion by beshoo** [laravel/nova-issues#5703] [General]
Hi.
Am looking to export the indexQuery. not the module data.
Please check my InsexQuery

```
 public static function indexQuery(NovaRequest $request, $query): \Illuminate\Database\Eloquent\Builder
    {
        $user = $request->user();
        $query->leftJoin('attendeds', function ($join) {
            $join->on('attendeds.user_id', '=', 'users.id')
                ->whereNotNull('attendeds.presence')
                ->whereNotNull('attendeds.leave');
        })
            ->select('users.id', 'users.email', 'users.name', 'users.phone', 'users.password', 'users.time_zone',
                DB::raw('SUM(CASE WHEN attendeds.presence IS NOT NULL AND attendeds.leave IS NOT NULL THEN TIME_TO_SEC(TIMEDIFF(attendeds.leave, attendeds.presence)) ELSE 0 END) as total_attended'))
            ->groupBy('users.id', 'users.email', 'users.name', 'users.phone', 'users.password', 'users.time_zone');

        if ($user->isSuperAdmin() or $request->user()->hasRole('admin')) {
            return $query;
        } else {
            return $query->where('users.id', $user->id);
        }
```
How can we export what the user really says in the Nova dashboard. since I altered the viewed data?

Regards

**Comment by crynobone** [comment]
We don't recommend using JOIN in indexQuery. This will interfere with some Resource default behaviour.

**Comment by beshoo** [comment]
I understand that you want me to create a lens for that. However, how can we export what the user really sees on the page, rather than just the module table?

**Reply by crynobone** [reply]
![image](https://github.com/laravel/nova-issues/assets/172966/8d20ab30-a473-452d-8e21-f9812afd8ef3)

https://nova.laravel.com/docs/4.0/resources/#csv-export

**Reply by beshoo** [reply]
I understand your concern. While creating a join, you were able to generate an additional calculated value named "total_attended", which cannot be obtained directly from the module. My question is, how can we export this data along with the existing information? Also, what is the best way to access the results of the quarry?

**Reply by beshoo** [reply]
I have come up with another idea. Is it possible to modify the Laravel module data before it is received by Nova? This way, Nova will recognize it as module data and display a preview, allowing us to export it to CSV without using WithFormat.o CSV without using WithFormat.

**Comment by beshoo** [comment]
Solved by using Lenses
